toolchain/ltrace.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM.
+# Author: Jayarama<jsomalra@linux.vnet.ibm.com>
+
+# Based on code by
+#   Author: Abdul Haleem <abdhalee@linux.vnet.ibm.com>
+
+
+import os
+import fnmatch
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+from avocado.utils import git
+from avocado.utils import distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Ltrace(Test):
+
+    def setUp(self):
+
+        backend = SoftwareManager()
+        dist = distro.detect()
+
+        """
+        Install  all the required dependencies
+        Building source tarball requires packages specific to os
+        that needs to be installed, if not installed test will stop.
+        """
+        if not backend.check_installed("gcc") and not backend.install("gcc"):
+            self.error("gcc is needed for the test to be run")
+
+        pkgs = ['git', 'wget', 'rpm', 'autoconf', 'automake',
+                'dejagnu', 'binutils']
+
+        if dist.name == 'sles':
+            sles_deps = ['build', 'libdw-devel', 'libelf-devel',
+                         'elfutils', 'binutils-devel', 'libtool', 'gcc-c++']
+            pkgs += sles_deps
+
+        elif dist.name == ('redhat'):
```

Comment:
The parenthesis are not necessary. My recommendation was (and still is) to use:

```
    ...
elif dist.name in ("redhat", "fedora"):
    ...
elif dist.name == "ubuntu":
    ...
```

Parenthesis there would be necessary as you'd be looking for the `dist.name` in tuple.